StaticInstaller.php

```diff
@@ -111,10 +121,32 @@ public static function composerFinishUpdate(Event $composer_event = null)
                 $text .= '-> Updating app "' . $app_alias . '": ' . ($result ? $result : 'Nothing to do') . ".\n";
                 self::printToStdout($text);
             }
-            unset($temp['update']);
+        }
+        if ($temp['update']!=null){
+            $updatedPackages = $temp['update'];
+        }
+        self::printToStdout("Unlink unused backup components:\n");
+        $installer = new self();
+        $apps = $installer->getAllApps();
+        $backupTime = $temp['backupTime'];
+        $unlinkResult = array();
+
+        foreach($apps as $app){
+
+            if (!in_array($app,self::EXCLUDE_BACKUP_PACKAGES) &&
+                !in_array($app,$updatedPackages)){
+                $unlinkResult[] = $installer->unlinkBackup($app,$backupTime);
+            }
+        }
+        $installer->copyTempFile($backupTime);
+        unset($temp['update']);
+        if (!in_array(false,$unlinkResult)){
+            self::printToStdout("Cleared backup from excess data. \n");
             self::setTempFile($temp);
         }
-        
+        else {
+            self::printToStdout("Could not clear backup. Please unlink manually. Check LastInstall.temp.json for list of apps to keep. \n");
```

Comment:
Kannst du hier bitte noch den Pfad zum betroffenen Backup ausgeben. Idealerweise also: "Could not clean up backup in xxx/xxx/xxx. Please unneeded backups manually if needed." Bitte logge auch eine Exception an dieser Stelle (es ist ja ein Fehler): $installer->getWorkbench()->getLogger()->logExceptions(new RuntimeException). Wir müssen diese Fehler ja irgendwie mitbekommen und das geht nur über das Log.

Eigentlich wäre noch eleganter, wenn du in der Unlink-Methode eine Exception wirfst und sie hier auffängst. Dann hättest du sogar eine Möglichkeit nach Fehlerursachen zu unterscheiden und hättest gleich was zu loggen. Die suche nach eine FALSE in einem array und dann auch noch ohne Strict-Mode ist nicht besonders sicher.

Reply:
Hallo Andrej,
die entsprechende Fehlermeldung gibt es schon in der Funktion "unlinkBackup". Ich habe jetzt eine Exception hinzugefügt. Die letzte Meldung ist nur nochmal abschließend als Fazit, dass Dinge schief gelaufen sind beim Unlink - als zusätzliche Aufforderung, dass man sich das Thema mal anschauen sollte und was man überhaupt zu tun hat.